docs/madr/decisions/011-multizone-envoy-admin.md

```diff
@@ -0,0 +1,120 @@
+# Multizone Envoy Admin Operations
+
+* Status: accepted
+
+Technical Story: https://github.com/kumahq/kuma/issues/5080
+
+## Context and Problem Statement
+
+Some time ago we introduced functionality that lets user execute Config Dump / Stats / Clusters from the CP.
+This is also available through Global CP in multi zone deployment.
+The problem is that Zone CP from given zone connects to only one instance of Global CP and admin operations are piped through KDS.
+In HA deployment, when you execute the Admin Operation request you don't know which instance you will use.
+
+## Considered Options
+
+* Discovery
+  * Config
+* Communication
+  * API Server
+  * Inter-CP Server
+
+## Pros and Cons of the Options
+
+We need to pass the request to a proper instance, for this we need to consider how we want to communicate between CP instances and how they will discover each other.
+
+## Discovery
+
+### Config
+
+We could gather all instances and IPs in one Config object (ConfigMap on Kubernetes).
+```yaml
+type: Config
+name: cp-instances
+config: |
+  {
+    "instances": [
+      {
+        "address": "192.168.1.100",
+        "apiServerHttpsPort": 5682,
+        "id": "<instance_id>"  
+      }
+    ]
+  }
+```
+
+Every CP instance would start a component that updates itself in the instances list if they are not on the list.
+Additionally, a leader would check the connection to all instances and remove them from the list. This way we can clear dead instances without using heartbeat, avoiding time based events, clock skews etc.
+On Kubernetes, we could have an alternative implementation. Leader can start a component that fetches Pods in system namespace and dumps this to a Config.
+However, I'd like to avoid separate implementations for Kubernetes and Universal.
+
+**How to retrieve IP?**
+Each CP needs to know its IP address. We can go over network interfaces on the host and take the first which is not localhost.
+This comes with a cost that we can pick incorrect interface, therefore we also need a config to override this explicitly.
```

Comment:
We can allow setting this through `env` or CLI. In that case on K8s we can take it from `status.podIP` in deployment config:
```
env:
  - name: POD_IP
    valueFrom:
      fieldRef:
        apiVersion: v1
        fieldPath: status.podIP
```

Reply:
Yes, although this is not a problem at all on Kubernetes, because in the container you have only one network interface

Reply:
I think using CNI you can inject additional network interfaces if you need. At least that's what Network Service Mesh project is doing :)

Reply:
Ok, we can also do explicit `POD_IP`. As you pointed out, it's pretty straightforward 👍

Reply:
How do this work in universal?

Reply:
a new static CP config.

Reply:
Yes